### PR TITLE
[ODBC] Support ODBC Sink for insert into data to ODBC external table

### PR DESCRIPTION
**Pull request by HappenLee**
issue:#5031

1. Refactoring code of ODBC scanner to ODBC connecter to ODBC Sink for insert into data to ODBC external table.
2. Support Transaction for ODBC sink to make sure insert into data is atomicital.
3. The document about ODBC sink has been modified

## Proposed changes

Describe the big picture of your changes here to communicate to the maintainers why we should accept this pull request. If it fixes a bug or resolves a feature request, be sure to link to that issue.

## Types of changes

What types of changes does your code introduce to Doris?
_Put an `x` in the boxes that apply_

- [] Bugfix (non-breaking change which fixes an issue)
- [x] New feature (non-breaking change which adds functionality)
- [] Breaking change (fix or feature that would cause existing functionality to not work as expected)
- [x] Documentation Update (if none of the other choices apply)
- [x] Code refactor (Modify the code structure, format the code, etc...)

## Checklist

_Put an `x` in the boxes that apply. You can also fill these out after creating the PR. If you're unsure about any of them, don't hesitate to ask. We're here to help! This is simply a reminder of what we are going to look for before merging your code._

- [x] I have create an issue on (Fix #ISSUE), and have described the bug/feature there in detail
- [x] Compiling and unit tests pass locally with my changes
- [x] I have added tests that prove my fix is effective or that my feature works
- [x] If this change need a document change, I have updated the document
- [x] Any dependent changes have been merged

### CODE REVIEW

File: be/src/exec/odbc_connecter.cpp [hunk 1]
```diff
@@ -172,7 +180,175 @@ Status ODBCScanner::get_next_row(bool* eos) {
...
+    }
+
+    return Status::OK();
+}
+
+Status ODBCConnecter::insert_row(const std::string& table_name, TupleRow *row) {
+    std::stringstream ss;
+
+    // Construct Insert statement of mysql
+    ss << "INSERT INTO " << table_name << " VALUES (";
```
Comment (yangzhg) [inline be/src/exec/odbc_connecter.cpp:211]: if use mysql statement,  table name better to surrounded by ```

File: be/src/exec/odbc_connecter.cpp [hunk 2]
```diff
@@ -172,7 +180,175 @@ Status ODBCScanner::get_next_row(bool* eos) {
...
+                std::stringstream err_ss;
+                err_ss << "can't convert this type to mysql type. type = " <<
+                       _output_expr_ctxs[i]->root()->type();
+                return Status::InternalError(err_ss.str());
+            }
+        }
+    }
+    ss << ")";
+
+    // Translate utf8 string to utf16 to use unicode codeing
```
Comment (yangzhg) [inline be/src/exec/odbc_connecter.cpp:306]:
```suggestion
    // Translate utf8 string to utf16 to use unicode encoding
```

File: be/src/exec/odbc_connecter.h [hunk 3]
```diff
@@ -52,32 +58,46 @@ struct DataBinding : public boost::noncopyable {
     ~DataBinding() { free(target_value_ptr); }
 };
 
-// ODBC Scanner for scan data from ODBC
-class ODBCScanner {
+// ODBC Connecter for scan data from ODBC
+class ODBCConnecter {
```
Comment (yangzhg) [inline be/src/exec/odbc_connecter.h:62]:
```suggestion
class ODBCConnector {
```

File: gensrc/thrift/DataSinks.thrift [hunk 4]
```diff
@@ -82,6 +83,12 @@ struct TMysqlTableSink {
     6: required string table
 }
 
+struct TOdbcTableSink {
+    1: optional string connect_string
+    2: optional string table
+    3: optional bool is_transaction
```
Comment (yangzhg) [inline gensrc/thrift/DataSinks.thrift:89]: maybe `use_transaction` is better

File: be/src/exec/odbc_connecter.cpp (hunk 1 above)
Comment (HappenLee) [reply be/src/exec/odbc_connecter.cpp:211]: I add the surround in FE

File: be/src/exec/odbc_connector.cpp [hunk 5]
```diff
@@ -48,18 +50,24 @@ static std::u16string utf8_to_wstring(const std::string& str) {
 
 namespace doris {
 
-ODBCScanner::ODBCScanner(const ODBCScannerParam& param)
+ODBCConnector::ODBCConnector(const ODBCConneterParam& param)
```
Comment (yangzhg) [inline be/src/exec/odbc_connector.cpp:53]:
```suggestion
ODBCConnector::ODBCConnector(const ODBCConnetorParam& param)
```

**Review by morningman — APPROVED** [review]
LGTM

**Pull request merged**